src/contracts/Protocol/ClaimController.sol

```diff
@@ -19,21 +59,24 @@ contract ClaimController is Ownable, IClaimController, OnlyApi {
         claimLimits[questionId] = claimLimit;
     }
 
-    function claim(uint256 questionId) public onlyOwner {
+    function claim(address user, uint256 questionId) public onlyApi {
         if (claims[questionId].length >= claimLimits[questionId]) revert ClaimLimitReached();
+        if (answers[questionId][user].author == user) revert AlreadyClaimed();
 
-        claims[questionId].push(_msgSender());
-        Answer memory _answer = Answer({state: STATE.CLAIMED, author: _msgSender(), answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
-        answers[questionId][_msgSender()] = _answer;
+        claims[questionId].push(user);
+        Answer memory _answer = Answer({state: STATE.CLAIMED, author: user, answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
```

Comment:
How about an enum for the Grade so we know the available grades/score - I don't think any unsigned integer should be allowed for a grade

Reply:
I like that idea, good call. Perhaps we could maybe cap it at 100 too. Will check what is most gas efficient in that too. However, I think that would be nice to iterate on and include in another PR. This PR mostly serves to make claims functional at the most basic level.

Reply:
Sure can you just add a TODO and then we can merge in an iterate off it?